R/verb-select.R

```diff
@@ -131,6 +131,28 @@ add_select <- function(lazy_query, vars) {
   )
 }
 
+select_can_be_inlined <- function(lazy_query, vars) {
+  vars_data <- op_vars(lazy_query)
+
+  bijective_or_not_distinct <- is_bijective_projection(vars, vars_data) ||
+      !is_true(lazy_query$distinct)
+
+  computed_columns <- purrr::pmap(
+    lazy_query$select,
+    function(name, expr, ...) if (is_quosure(expr)) name
+  ) %>%
+    purrr::compact()
+
+  order_vars <- purrr::map_chr(lazy_query$order_by, as_label)
+
+  all(
```

Comment:
Given there's only two clauses here, I think `&&` would be easier to understand.